### PR TITLE
attack_delay variable for items

### PR DESCRIPTION
**Pull request by unid15**
- ~~Attack delay of an item is now 0.6 + 0.15*w_class seconds (instead of flat 1sec)~~
- ~~1 wclass = 0.75sec, 3 wclass = 1.05sec, 5 wclass = 1.35sec~~
- ~~Energy weapons aren't affected by this and have 1 second delay when activated~~
- Added attack_delay variable that can change the item's attack delay

(Gun fire rate isn't affected)

### CODE REVIEW

**Comment by Probe1** [comment]
There's an existing meta that I don't see any complaints about so altering it needs discussion first.

Gonna need a balance tag and discussion on this one.  I like the idea but this will require some heavy items to be rebalanced.  Fireaxe comes immediately to mind.

**Comment by Imagebard** [comment]
>Added attack_delay variable that can override the attack delay

Does this mean that we can manually set the attack delay for certain weapons?
*Did you set the executioner sword delay to be a bit longer than a item of the same w_class would be?*

**Comment by nuklearcellphoneg** [comment]
You're gonna need to balance all the items.

Muh machine gun screwdriver.

**Comment by 9600bauds** [comment]
Executioner's Sword teaches us that high damage per hit is much better than high DPS.

**Comment by ghost** [comment]
@Imagebard Probably not, actually. We'll need a test server to have a look though.

**Comment by Imagebard** [comment]
@9600bauds That doesn't mean its a good idea to ignore something's DPS.

**Comment by AutismSpider** [comment]
Executioner sword not going mean much when some one goes fist of the north start on you with a welding tool (does both burn and brute) or screwdriver to eyes/face

**Comment by Shadowmech88** [comment]
Is this going to turn into a STR vs DEX argument?

**Comment by Imagebard** [comment]
In SS13 close quarters, STR is a clear winner.

**Comment by clusterfack** [comment]
Going to be an issue with the items that have w_class 200 :^)

**Comment by Probe1** [comment]
What happens if I drop an item after attacking then pick it up again?

What happens if I attack with a slow item then attack with a fast item?

**Comment by ihadtoregisterforthis** [comment]
What happens if I drop an item after attack and then someone picks it up again?

Can I combo a fast hit with a throw then pick up then loop?

**Comment by Imagebard** [comment]
>What happens if I drop an item after attacking then pick it up again?
What happens if I attack with a slow item then attack with a fast item?

You can't attack with a fast item until the cooldown for the slow item ends.

>What happens if I drop an item after attack and then someone picks it up again?

He can swing the item instantly.
People, this only changes the item swing delay. Its not changing the actual attack code. Those things you mentioned will be the same even if this is merged. Why are you asking those unrelated questions?

**Comment by Probe1** [comment]
Yeah but what if I attack with a slow weapon, drop it, and pick it back up then attack again?  I think that's a pretty big speed bump for the code and I want to know if that will be incorporated in the design.

**Comment by Shadowmech88** [comment]
No matter what you do, you're unable to attack again, with anything, until the cooldown from the last weapon with with you attacked ends.

**Comment by Probe1** [comment]
I'm satisfied!

:+1: 

Course this still opens a can of worms and @unid15 I want you to be ready to pitch in and help us sort out the new balancing that will come up.  But I'm ok with going forward.

**Comment by Imagebard** [comment]
>I'm satisfied!
:+1:

Why are you +1'ing this PR.
This PR doesn't change anything about swing delay.

**Comment by Probe1** [comment]
>asking a man who willingly became an admin if he has brain damage.

Of course I do.

The addition opens new avenues for combat and content.  It doesn't overwhelmingly disrupt the current game and it doesn't LOOK broken.  But I am a very bad judge of that.

**Comment by PJB3005** [comment]
@Probe1 I think @Imagebard ment that you should use GitHub's new reactions system.

**Comment by Imagebard** [comment]
No, I'm asking why he gave a :+1: to a PR he disliked. After asking a bunch of unrelated questions. It just makes no sense.

**Comment by Probe1** [comment]
I dislike a lot of things that get merged and it isn't my goddamn business to exert total control over the creative efforts of strangers for the shared good of us all.

The questions weren't unrelated, they were all tied to whether this was fundamentally going to work or wasn't going to work.  If a fireaxe has an attack speed of 10 but I can just drop it and pick it back up in half the time then it aint gonna work.  I've been told it will work, so why not?

**Comment by clusterfack** [comment]
Its going to take 5 minutes to assemble a mech.

**Comment by FudgePucker** [comment]
Oh boy tool box is even more worthless compared to welding tool or glass shard. Great pr

**Comment by AutismSpider** [comment]
@FudgePucker  if does this get in just toss glass shards at the wall until you get  damage one or make a spear

**Comment by clusterfack** [comment]
I have to agree with one thing, the current force system is balanced entirely around everything having the same attack delay.

**Comment by Probe1** [comment]
I thought tools didn't have any sort of delay regardless when they weren't used on mobs

**Comment by unid15** [comment]
needs some balancing first

**Comment by clusterfack** [comment]
@Probe1 No everything in attackby has a delay, its just that tools used to have an even bigger delay

**Comment by Probe1** [comment]
i c

**Comment by unid15** [comment]
Removed everything except for the attack_delay variable - so there are no gameplay changes

**Comment by ghost** [comment]
Chicken.

**Comment by unid15** [comment]
I'm doing this later, I reopened this just to put the attack_delay var into the game

**Comment by 9600bauds** [comment]
>Chicken

Space not cold when

**Comment by nuklearcellphoneg** [comment]
but why is space cold

**Comment by Probe1** [comment]
because they don't have enough manncrowaves in spess

**Comment by nuklearcellphoneg** [comment]
I don't understand the dislike towards that thing, it seems quicker than just shaking someone up, and...

O-oh that's not the issue here.

**Comment by Shadowmech88** [comment]
Even if it is quicker than shaking them up, there's no reason to have a new machine for that.
Case in point: the sleepers now have a "wake up occupant" option that functions just the same.

**Comment by Shadowmech88** [comment]
When's this getting merged, anyway?

File: code/_onclick/click.dm [hunk 1]
```diff
@@ -82,18 +82,22 @@
 		return
 
 	var/obj/item/W = get_active_hand()
+	var/item_attack_delay = 0
+
+	if(W)
```
Comment (clusterfack) [inline code/_onclick/click.dm]: Why would this check be necessary instead of just having the next line after if(W == A)

**Comment by unid15** [comment]
I forgot that this exists

when

**Comment by clusterfack** [comment]
Now I guess, I'm not sure it will 100% work without reducing the attack delayer min delay but if not you can fix that later

**Pull request merged**